### ISSUE TITLE
v3.1.0 fails on iOS 8 deployment target (Carthage/Cocoapods)

### ISSUE THREAD

**Issue by dev4dev** [SnapKit/SnapKit#348]
### New Issue Checklist

🚫 If this template is not filled out your issue **will** be closed with no comment. 🚫

* [x] I have looked at the [Documentation](http://snapkit.io/docs)
* [x] I have read the [F.A.Q.](http://snapkit.io/faq)
* [x] I have filled out this issue template.

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform                | e.g. ios
 Platform Version        | e.g. 10.0
 SnapKit Version         | e.g. 3.1.0
 Integration Method      | e.g. carthage
 

### Issue Description

```
*** Checking out SnapKit at "3.1.0"
*** xcodebuild output can be found in /var/folders/fv/5yb6ypv92qd6rll9n1gpclzw0000gn/T/carthage-xcodebuild.A5ONmh.log
*** Building scheme "SnapKit" in SnapKit.xcworkspace
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
.../Carthage/Checkouts/SnapKit/Source/LayoutConstraintItem.swift:34:11: error: 'ConstraintLayoutGuide' is only available on iOS application extension 9.0 or newer
...Carthage/Checkouts/SnapKit/Source/LayoutConstraintItem.swift:34:11: error: 'ConstraintLayoutGuide' is only available on iOS application extension 9.0 or newer
A shell task (/usr/bin/xcrun xcodebuild -workspace .../Carthage/Checkouts/SnapKit/SnapKit.xcworkspace -scheme SnapKit -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

**Comment by robertjpayne** [comment]
Updated the title of this, this is an issue because 3.1.0 introduces code that will only compile when targeting iOS 9 or newer, it needs availability checks!

**Comment by kaandedeoglu** [comment]
@robertjpayne bumping iOS Deployment Target to 9.0 in Build Settings fixed the issue for me. I don't understand why it was failing in the first place, given that you're defining the class conditionally for iOS 9+

**Comment by robertjpayne** [comment]
@kaandedeoglu I'm probably missing a availability wrap somewhere.

**Comment by MacMeDan** [comment]
@dev4dev did you find a solution for this? I am having this same problem.

**Comment by dev4dev** [comment]
@MacMeDan nope, I just set version 3.0.2 and waiting for new release =)

**Comment by robertjpayne** [comment]
3.1.0 is a pre-release for a reason! Will fix it when I have time probably this weekend.

**Comment by timsearle** [comment]
Perhaps would make sense to keep pre-release versions on separate branches rather than tagged as releases? As calling `carthage update` does pull down this non-working version.

**Comment by robertjpayne** [comment]
@timsearle what do you have Carthage tagged to? I strongly recommend you pin your dependencies to hard versions with the way Swift and iOS break across deployment targets / Swift versions.

There is no way to "separate" out a pre-release so Carthage doesn't see it unless I just constantly call it "next" but then you can't convert it to a real release w/o a new tag either.

This is fixed in v3.1.1.
